From 170f9acfb41704b68e2ba98690fd6d5e98addd85 Mon Sep 17 00:00:00 2001
From: Eike Ziller <eike.ziller@qt.io>
Date: Thu, 24 Aug 2023 13:41:33 +0200
Subject: [PATCH] ClangTools: Add support for yaml-cpp 0.8
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

They renamed their target from `yaml-cpp`
to `yaml-cpp::yaml-cpp`

Still support < 0.8 though.

Fixes: QTCREATORBUG-29537
Change-Id: Iafa6e75998f0120610542b4c744b205aee3392a3
Reviewed-by: Qt CI Bot <qt_ci_bot@qt-project.org>
Reviewed-by: Lisandro Damián Nicanor Pérez Meyer <perezmeyer@gmail.com>
Reviewed-by: David Schulz <david.schulz@qt.io>
---
 cmake/Findyaml-cpp.cmake | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/cmake/Findyaml-cpp.cmake b/cmake/Findyaml-cpp.cmake
index e7b3d361d3..0b0e81418a 100644
--- a/cmake/Findyaml-cpp.cmake
+++ b/cmake/Findyaml-cpp.cmake
@@ -15,7 +15,15 @@ if (yaml-cpp_FOUND)
     unset(yaml_cpp_include_dir CACHE)
     find_path(yaml_cpp_include_dir yaml-cpp/yaml.h)
   endif()
-  set_target_properties(yaml-cpp PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${yaml_cpp_include_dir}")
+  if(TARGET yaml-cpp::yaml-cpp)
+    # yaml-cpp >= 0.8
+    set_property(TARGET yaml-cpp::yaml-cpp PROPERTY IMPORTED_GLOBAL TRUE)
+    add_library(yaml-cpp ALIAS yaml-cpp::yaml-cpp)
+    set(yaml-cpp_TARGET yaml-cpp::yaml-cpp)
+  else()
+    set(yaml-cpp_TARGET yaml-cpp)
+  endif()
+  set_target_properties(${yaml-cpp_TARGET} PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${yaml_cpp_include_dir}")
 else()
   if(TARGET yaml-cpp)
     return()
-- 
2.41.0

